Mutual Auth Docs (#25509)
cilium-team/cilium: Documentation/network/servicemesh/mutual-authentication/installation.rst

Prerequisites
#############

* Mutual authentication is only currently supported with SPIFFE APIs for certificate management. 

> * Users have the option to install a SPIRE server or they can also bring their own instead.

Reviewer: ```suggestion
* The Cilium Helm chart includes an option to deploy a SPIRE server for mutual authentication. You may also deploy your own SPIRE server and configure Cilium to use it.
```

  
Installation
############

.. tabs::

    .. group-tab:: Helm

>         Mutual authentication can be enabled with helm flag ``authentication.mutual.spire.enabled``

Reviewer: ```suggestion
        Mutual authentication can be enabled with the helm flag ``authentication.mutual.spire.enabled``
```

        set as true. 

>         A SPIRE server can be deployed alongside it with the helm

Reviewer: Can we change the wording here a bit? We're saying that the SPIRE server can be deployed alongside mutual auth, but mutual auth seems to be more of an option rather than something that can be deployed. For instance, maybe something like: "A manager SPIRE server can be deployed for you by setting the helm flag...."

        flag ``authenticaton.mutual.spire.install.enabled`` also set as true.

>         Please refer to :ref:`k8s_install_helm` for a fresh installation.

Reviewer: Can you expand on what you mean by a fresh installation? I think this is a bit inconsistent, as in the Cilium CLI tab, we use "cilium install" which is for a fresh install, but here we use "helm upgrade" and assume the user has already installed cilium.

        .. parsed-literal::

            $ helm upgrade cilium |CHART_RELEASE| \\
                --namespace kube-system \\
                --reuse-values \\
                --set authentication.mutual.spire.enabled=true \\
                --set authentication.mutual.spire.install.enabled=true

            $ kubectl -n kube-system rollout restart deployment/cilium-operator
            $ kubectl -n kube-system rollout restart ds/cilium

        

>         Next you can check the status of the Cilium agent and operator:

Reviewer: Can we expand on this line to explain that this should be done to validate that the installation was successful? This is to match the type of style used in the Cilium install docs at https://docs.cilium.io/en/stable/installation/k8s-install-helm/. This applies to [1] as well.

        .. code-block:: shell-session

            $ cilium status

>         .. include:: ../../../installation/cli-download.rst

Reviewer: Can we move this include above the call to `cilium status`, so the user gets the installation instructions and then the usage instructions?

    .. group-tab:: Cilium CLI

        .. include:: ../../../installation/cli-download.rst

        Mutual authentication and associated SPIRE server can be enabled with the following command. 

>         Note this requires the Cilium CLI Helm mode (enabled with setting the environment variable with ``CILIUM_CLI_MODE=helm``).

Reviewer: Can we also add a note that this requires Cilium CLI version >= v0.15.0?

Reviewer 2: Does it? I thought the Helm mode was available opt-in a bit before that. But maybe easiest to just stick to 0.15.0 indeed.

Author: Helm mode became default in 0.15.0 but available in 0.14.0. 

As the docs are being updated with the assumption of v0.15, I will do the same here.

        .. code-block:: shell-session

>             $ cilium install \\
>                 --helm-set authentication.mutual.spire.enabled=true \\

Reviewer 2: You need the double backslashes in the `.. parse-literal::` blocks, because backslashes are parsed as RST escape markers there, but I think you need just one per line in the `.. code-block::`. Please double-check on the preview.

```suggestion
            $ cilium install \
                --helm-set authentication.mutual.spire.enabled=true \
```

                --helm-set authentication.mutual.spire.install.enabled=true

>         Next you can check the status of the Cilium agent and operator:

Reviewer: [1]

        set as true. 
        A SPIRE server can be deployed alongside it with the helm
        flag ``authenticaton.mutual.spire.install.enabled`` also set as true.
        Please refer to :ref:`k8s_install_helm` for a fresh installation.

        .. parsed-literal::

            $ helm upgrade cilium |CHART_RELEASE| \\
                --namespace kube-system \\
                --reuse-values \\
                --set authentication.mutual.spire.enabled=true \\
                --set authentication.mutual.spire.install.enabled=true

            $ kubectl -n kube-system rollout restart deployment/cilium-operator
            $ kubectl -n kube-system rollout restart ds/cilium

        
        Next you can check the status of the Cilium agent and operator:

        .. code-block:: shell-session

            $ cilium status

        .. include:: ../../../installation/cli-download.rst

    .. group-tab:: Cilium CLI

        .. include:: ../../../installation/cli-download.rst

        Mutual authentication and associated SPIRE server can be enabled with the below command.

        .. code-block:: shell-session

>             $ cilium install \\

Reviewer 3: @nvibert  Might be worth a note that this requires the Cilium CLI Helm mode enabled (`CILIUM_CLI_MODE=helm`). Otherwise, SPIRE will not be installed.

Prerequisites
#############

> * Mutual authentication is only currently supported with SPIFFE APIs. 

Reviewer 4: ```suggestion
* Mutual authentication is only currently supported with SPIFFE APIs for certificate management. 
```

> * By default, a SPIRE server is deployed when Cilium is installed.

Reviewer 4: is it? I remember it might not be in a last discussion? Better to double check this

Author: Ahhh yes you're right - I had changed it to default when it looked like it was going to be there by default but I need to change it back 😅
